public/app/features/explore/state/selectors.ts

```diff
@@ -28,3 +31,43 @@ export const deduplicatedLogsSelector = createLodashMemoizedSelector(
     return dedupLogRows(filteredData, dedupStrategy);
   }
 );
+
+const parseRawTime = (value): Moment | string => {
+  if (value === null) {
+    return null;
+  }
+
+  if (value.indexOf('now') !== -1) {
+    return value;
+  }
+  if (value.length === 8) {
+    return moment.utc(value, 'YYYYMMDD');
+  }
+  if (value.length === 15) {
+    return moment.utc(value, 'YYYYMMDDTHHmmss');
+  }
+  // Backward compatibility
+  if (value.length === 19) {
+    return moment.utc(value, 'YYYY-MM-DD HH:mm:ss');
+  }
+
+  if (!isNaN(value)) {
+    const epoch = parseInt(value, 10);
+    return moment.utc(epoch);
+  }
+
+  return null;
+};
+
+export const timeRangeFromUrlSelector = (range: RawTimeRange, timeZone: TimeZone): TimeRange => {
+  const raw = {
+    from: parseRawTime(range.from),
```

Comment:
Need to do the parsing here since when going to Explore from dashboard `updateLocation` action is dispatched and it feld bad to do this parsing in the `updateLocation` reducer logic - should have no side effects right?